### PR TITLE
Ref #42068 Hide 'preferred_spaces' field on create employee form. Fix…

### PR DESCRIPTION
**Pull request by boddy-devbranch**
…ed options of available roles for current user.

### CODE REVIEW

**Review by andrew-sereda — APPROVED** [review]

File: modules/living_spaces_intranet/living_spaces_intranet.module [hunk 1]
```diff
@@ -255,3 +255,21 @@ function living_spaces_intranet_employee_form_submit($form, FormStateInterface $
     ]));
   }
 }
+
+/**
+ * Implements hook_form_BASE_FORM_ID_alter().
+ */
+function living_spaces_intranet_form_user_form_alter(&$form, FormStateInterface $form_state, $form_id) {
+
+  if (isset($form['account']['roles']['#options']) && \Drupal::hasService('delegatable_roles')) {
```
Comment (andrew-sereda) [inline modules/living_spaces_intranet/living_spaces_intranet.module]: You have not added module with this service to composer.json in profile

**Review by andrew-sereda — CHANGES_REQUESTED** [review]

File: config/install/core.entity_form_display.user.user.employee.yml [hunk 2]
```diff
@@ -12,6 +12,8 @@ dependencies:
     - image
     - profile
     - user
+_core:
+  default_config_hash: _W4c-Pjk06XeQX4Q-UnsEoLwAfSygIDBGiUiPz1LJIg
```
Comment (victor-aston) [inline config/install/core.entity_form_display.user.user.employee.yml]: no need to add that info.

File: modules/living_spaces_intranet/living_spaces_intranet.module (hunk 1 above)
Comment (victor-aston) [inline modules/living_spaces_intranet/living_spaces_intranet.module]: you should add the module as a dependency to the info file and remove `\Drupal::hasService()` validation.

File: modules/living_spaces_intranet/living_spaces_intranet.module [hunk 3]
```diff
@@ -255,3 +255,21 @@ function living_spaces_intranet_employee_form_submit($form, FormStateInterface $
...
+    // Set options of available roles for current user.
+    $current_user = \Drupal::currentUser();
+    $assignableRoles = \Drupal::service('delegatable_roles')
+      ->getAssignableRoles($current_user);
+    if (isset($form['account']['roles']['#options']['authenticated'])) {
+      $assignableRoles['authenticated'] = $form['account']['roles']['#options']['authenticated'];
+    }
+    $form['account']['roles']['#options'] = array_intersect($form['account']['roles']['#options'], $assignableRoles);
```
Comment (victor-aston) [inline modules/living_spaces_intranet/living_spaces_intranet.module]: please, use `$assignable_roles` instead of `$assignableRoles`

**Review by victor-aston — CHANGES_REQUESTED** [review]
please, fix some small issues

**Review by victor-aston — APPROVED** [review]

**Review by andrew-sereda — APPROVED** [review]

**Pull request merged**